### PR TITLE
Refactor rerank tei

### PR DESCRIPTION
**Pull request by WenjiaoYue**
## Description

Refactor the TEI-related part in rerank.

## Issues

 `n/a`.

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [x] Breaking change (fix or feature that would break existing design and interface)
- [ ] Others (enhancement, documentation, validation, etc.)

## Dependencies
no

## Tests

CI pass.

### CODE REVIEW

File: comps/reranks/src/integrations/opea_tei_reranking.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
+# Copyright (C) 2024 Intel Corporation
```
Comment (chensuyue) [inline comps/reranks/src/integrations/opea_tei_reranking.py:1]: `opea_tei_reranking.py` -> `opea_tei.py`

File: tests/reranks/test_reranks_tei.sh [hunk 2]
```diff
@@ -6,14 +6,15 @@ set -x
```
Comment (chensuyue) [inline tests/reranks/test_reranks_tei.sh:6]: `test_reranks_tei.sh` -> `tests/reranks/test_reranks_opea_tei.sh`

**Review by letonghan — APPROVED** [review]
lgtm

File: comps/reranks/src/integrations/opea_tei_reranking.py (hunk 1 above)
Comment (WenjiaoYue) [reply comps/reranks/src/integrations/opea_tei_reranking.py]: DONE

File: tests/reranks/test_reranks_tei.sh (hunk 2 above)
Comment (WenjiaoYue) [reply tests/reranks/test_reranks_tei.sh:6]: DONE

**Pull request merged**